### ISSUE TITLE
pnpm安装依赖报 Forbidden - 403

### ISSUE THREAD

**Issue by SuperBug-RhysLiu** [pnpm/pnpm#6445]
i need help and I don't understand why this problem arises
pnpm安装某些私有仓库的依赖报403
类似于以下依赖,包括但不止一个
 ERR_PNPM_FETCH_403  GET https://xxxxxx/@commonbox%2futils/-/utils-1.4.0.tgz: Forbidden - 403
No authorization header was set for the request.
这样的问题,在多个pnpm的版本中出现,但是,在pnpm@6.32.23这个版本中是好的..看不出什么问题
node 涉及12-18;
报错的依赖使用npm安装可以安装

**Comment by capndave** [comment]
Are you using a package repository other than npm or yard (like Artifactory, Nexus, etc.)? If so, please check with engineers at your company to correctly configure npm and your `.npmrc` file

**Comment by scl930227** [comment]
> Are you using a package repository other than npm (like Artifactory, Nexus, etc.)? If so, please check with engineers at your company to correctly configure npm and your `.npmrc` file

yes, 我们确实使用了npmrc, 后来联系开发工程师,发现该依赖是有权限的,但是npm下是好的,pnpm 在请求时缺少了heardes 下的某个鉴权属性,有什么方法可以让我在安装依赖的时候带上该属性吗

**Comment by iliuxudong** [comment]
> > Are you using a package repository other than npm (like Artifactory, Nexus, etc.)? If so, please check with engineers at your company to correctly configure npm and your `.npmrc` file
> 
> yes, 我们确实使用了npmrc, 后来联系开发工程师,发现该依赖是有权限的,但是npm下是好的,pnpm 在请求时缺少了heardes 下的某个鉴权属性,有什么方法可以让我在安装依赖的时候带上该属性吗

这个问题你解决了吗？

**Comment by zkochan** [comment]
Closing: this is not something pnpm can fix.

The private registry rejects the request as unauthorized, which is a matter of the registry's configuration and the credentials in the npmrc.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
